submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,304 @@
+                        View Profile
+                    </div>
+                </div>
+            </div>
+        </div>
+    `;
+}
+
+const renderUser = (user) => {
+    let picture = user.picture.medium;
+    let name = user.name.first + ' ' + user.name.last;
+    let age = user.dob.age;
+    let birthDate = new Date(user.dob.date).toLocaleDateString('en-us', { weekday:"long", year:"numeric", month:"short", day:"numeric"});
+    let country = user.location.country;
+    let state = user.location.state;
+    let city = user.location.city;
+    let email = user.email;
+    let phone = user.phone;
+    let gender = user.gender;
+
+    mainContainer.innerHTML = `
+        <div class="user user_single">
+            <div class="user-block">
+                <div class="user-poster">
+                    <img src="${picture}" alt="${name}">
+                </div>
+
+                <div class="user-headings">
+                    <h5><strong>${name}</strong></h5>
+                    <h6>${age} years old</h6>
+                </div>
+
+                <div class="user-description">
+                    <p><strong>User Location:</strong></p>
+                    <p>${birthDate}</p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Location:</strong></p>
+                    <p>${country}</p>
+                    <p>${state}</p>
+                    <p>${city}</p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Contacts:</strong></p>
+                    <p>
+                        <a href="mailto:${email}">${email}</a>
+                    </p>
+                    <p>
+                        <a href="tel:${phone}">${phone}</a>
+                    </p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Gender:</strong></p>
+                    <p class="uppercase">${gender}</p>
+                </div>
+
+                <div class="user-button user-button_close">
+                    Back to Friends
+                </div>
+            </div>
+        </div>
+    `;
+}
+
+// filters
+const sidebarFilters = document.querySelector('.main__sidebar');
+sidebarFilters.addEventListener("submit", function(event) {
+    event.preventDefault();
+
+    renderUsers(getSidebarOptions(), usersList);
+
+    if (window.screen.width <= 992) {
+        toggleMenu();
+    }
+});
+
+const getSidebarOptions = () => {
+    const formData = new FormData(sidebarFilters);
+    const filtersObject = {};
+    formData.forEach((value, key) => {
+        filtersObject[key] = value;
+    });
+
+    return filtersObject;
+}
+
+const filterUsers = (filtersObject, usersArray) => {
+    const pipes = [filterByName, filterByGender, sortBy];
+    const filterName = filtersObject.name;
+    const filterGender = filtersObject.gender;
+    const filterSort = filtersObject.filterSort;
+
+    let users = pipes.reduce((list, func) => {
```

Comment:
Why not just do `usersFilteredList = pipes....` here? And then return it. What is the purpose of the `users` variable?

Reply:
Changed to usersFilteredList = pipes...., but I set usersFilteredList to be as let variable. Cause I'm overriding this value in a few places.